lib/vdc/BOVCollection.cpp

```diff
@@ -217,21 +249,37 @@ int BOVCollection::_validateParsedValues()
         _brickSizeAssigned = true;
     }
 
-    // Validate endian type
-    if (_tmpDataEndian != _dataEndian && _dataEndianAssigned == true)
-        return _inconsistentValueError(ENDIAN_TOKEN);
-    else {
-        _dataEndian = _tmpDataEndian;
-        _dataEndianAssigned = true;
-    }
-
     // Validate byte offest
     if (_tmpByteOffset != _byteOffset && _byteOffsetAssigned == true)
         return _inconsistentValueError(OFFSET_TOKEN);
     else {
         _byteOffset = _tmpByteOffset;
         _byteOffsetAssigned = true;
     }
+
+    // If _dataFile is not an absolute path, prepend with the BOV header's path
+    if (!Wasp::FileUtils::IsPathAbsolute(_dataFile)) { _dataFile = _currentFilePath + "//" + _dataFile; }
+
+    // Validate whether we can open the data file
+    FILE *fp = fopen(_dataFile.c_str(), "rb");
+    if (fp == nullptr) return _invalidFileError();
+
+    // Validate the data file's size
+    int    formatSize = _sizeOfFormat(_dataFormat);
+    size_t count = _gridSize[0] * _gridSize[1] * _gridSize[2];
+    auto   readBuffer = std::unique_ptr<unsigned char[]>(new unsigned char[count * formatSize]);
+    size_t numElements = fread(readBuffer.get(), formatSize, count, fp);
+    if (numElements != count) {
+        fclose(fp);
+        return _invalidFileSizeError(numElements);
+    }
```

Comment:
Scott, 2 comments:
1) when you validate the file size, remember to take into consideration the offset bytes.
2) your current implementation reads the entire file just to count its size. This is very cumbersome. You should use `std::fseek()` to try to access the desired location and then use the return value to determine if good or bad. 
https://en.cppreference.com/w/cpp/io/c/fseek

Reply:
should check return value of `fseek()` in other places too.